lib/client.js

```diff
@@ -51,26 +52,9 @@ extend(Raven.prototype, {
 
     // autoBreadcrumbs: true enables all, autoBreadcrumbs: false disables all
     // autoBreadcrumbs: { http: true } enables a single type
-    // this procedure will ensure that this.autoBreadcrumbs is an object populated
-    // with keys -> bools reflecting actual status of all breadcrumb types
-    var autoBreadcrumbDefaults = {
-      console: false,
-      http: false,
-    };
+    this.autoBreadcrumbs = options.autoBreadcrumbs || false;
     // default to 30, don't allow higher than 100
     this.maxBreadcrumbs = Math.max(0, Math.min(options.maxBreadcrumbs || 30, 100));
-    this.autoBreadcrumbs = extend({}, autoBreadcrumbDefaults);
```

Comment:
For which part in particular? I basically moved the "figure out what our autobreadcrumb to turn on" bit into instrumentor. I don't think anything really changed behavior-wise, just where it lives, so now client has to care a lot less and just has to pass its config on to the instrumentor. It actually ended up simplifying things quite a bit from a combination of two or three small differences.